.github/workflows/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    env:
+      GOFLAGS: -mod=readonly
+
+    steps:
+      - name: Set up Go
```

Comment:
possibly use a golang based image and skip this step

Reply:
See my point above. This is kind of standard for GitHub Actions. Most of the time it will come from host cache (from the machine) anyway.